beefy-gadget/src/round.rs

```diff
@@ -53,51 +58,51 @@ fn threshold(authorities: usize) -> usize {
 	authorities - faulty
 }
 
-pub(crate) struct Rounds<Hash, Number, Id, Signature> {
-	rounds: BTreeMap<(Hash, Number), RoundTracker<Id, Signature>>,
-	validator_set: ValidatorSet<Id>,
+pub(crate) struct Rounds<H, N, P>
+where
+	P: Pair,
+	P::Public: AppPublic,
+	P::Signature: Clone + Codec + Debug + PartialEq + TryFrom<Vec<u8>>,
+{
+	rounds: BTreeMap<(H, N), RoundTracker<P::Public, P::Signature>>,
+	validator_set: ValidatorSet<P::Public>,
 }
 
-impl<Hash, Number, Id, Signature> Rounds<Hash, Number, Id, Signature>
+impl<H, N, P> Rounds<H, N, P>
 where
-	Hash: Ord,
-	Number: Ord,
+	H: Ord,
+	N: Ord,
```

Comment:
I know it's a never ending discussion and we are not super consistent with this, but I prefer full names for generic parameters instead of one-letter's. It's maybe a bit longer to write, but at least you know roughly what is expected.
I know that the typical counter-argument is that it may be confused with regular types, and I usually suggest something like `THash, TNumber` in such case.

Not going to die on this hill though, but just wanted to leave my 2 cents here.

Reply:
I know, we discussed this earlier ;-) A lot of preferences usually come down to which tooling (esp. editor) one is using. 

But I'm anyways more and more in favour of using associated types instead of generic type parameters wherever possible. And with associated types longer, more descriptive names are unanimous :-)